OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -1092,6 +1092,15 @@ internal static void UpgradeWeaponRules(ModData modData, int engineVersion, ref
 					}
 				}
 
+				// Rename BurstDelay to BurstDelays
+				if (engineVersion < 20170716)
+				{
+					if (node.Key == "BurstDelay")
+						node.Key = "BurstDelays";
+					if (node.Key == "-BurstDelay")
```

Comment:
We do have a few cases in our official mods where inherited properties are cleared/reset to default this way, but I guess you're right that it's unnecessary here, for BurstDelay specifically it's extremely unlikely.